pkg/cmd/pulumi/packagecmd/package_info.go

```diff
@@ -0,0 +1,283 @@
+	return strings.TrimSpace(summary)
+}
+
+func simplifyModuleName(resourceName string) (string, error) {
+	split := strings.Split(resourceName, ":")
+	if len(split) < 3 {
+		return "", fmt.Errorf("invalid resource name %q", resourceName)
+	}
+	moduleSplit := strings.Split(split[1], "/")
+	return split[0] + ":" + moduleSplit[0] + ":" + split[2], nil
+}
+
+func showModuleInfo(spec *schema.PackageSpec, moduleName string, stdout io.Writer) error {
+	fmt.Fprintf(stdout, "Module: %s:%s\n", spec.Name, moduleName)
+
+	resources := make(map[string]schema.ResourceSpec)
+	for res, spec := range spec.Resources {
+		simplifiedName, err := simplifyModuleName(res)
+		if err != nil {
+			return err
+		}
+		split := strings.Split(simplifiedName, ":")
+		if len(split) < 3 {
+			return fmt.Errorf("invalid resource name %q", res)
+		}
+		if split[1] != moduleName {
+			continue
+		}
+		resources[split[2]] = spec
+	}
+
+	if len(resources) == 0 {
+		return fmt.Errorf("module %q not found", moduleName)
+	}
+	fmt.Fprintf(stdout, "Resources: %d\n", len(resources))
+
+	fmt.Fprintln(stdout)
+	for _, name := range maputil.SortedKeys(resources) {
+		fmt.Fprintf(stdout, " - %s: %s\n", name, summaryFromDescription(resources[name].Description))
+	}
+	return nil
+}
+
+func showResourceInfo(spec *schema.PackageSpec, moduleName, resourceName string, stdout io.Writer) error {
+	fullResourceName := fmt.Sprintf("%s:%s:%s", spec.Name, moduleName, resourceName)
+
+	fmt.Fprintf(stdout, "Resource: %s\n", fullResourceName)
+
+	res, ok := spec.Resources[fullResourceName]
+	if !ok {
+		for name, r := range spec.Resources {
+			simplifiedName, err := simplifyModuleName(name)
+			if err != nil {
+				return err
+			}
+			if fullResourceName == simplifiedName {
+				res = r
+				ok = true
+				break
+			}
+		}
+	}
+	if !ok {
+		return fmt.Errorf("resource %q not found", fullResourceName)
+	}
+
+	fmt.Fprintln(stdout)
+	fmt.Fprintln(stdout, "Input Properties:")
+	for _, name := range maputil.SortedKeys(res.InputProperties) {
+		prop := res.InputProperties[name]
+		optionalStr := ""
+		if !slices.Contains(res.RequiredInputs, name) {
+			optionalStr = " (optional)"
+		}
+		typ, err := getType(spec, prop.TypeSpec)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(stdout, " - %s (%s%s): %s\n",
+			name, typ, optionalStr, summaryFromDescription(prop.Description))
+	}
+
+	fmt.Fprintln(stdout)
+
+	fmt.Fprintln(stdout, "Output Properties:")
+	for _, name := range maputil.SortedKeys(res.Properties) {
+		prop := res.Properties[name]
+		presentStr := ""
+		if slices.Contains(res.Required, name) {
+			presentStr = " (always present)"
+		}
+		typ, err := getType(spec, prop.TypeSpec)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(stdout, " - %s (%s%s): %s\n",
+			name, typ, presentStr, summaryFromDescription(prop.Description))
+	}
+	return nil
+}
+
+func getType(spec *schema.PackageSpec, prop schema.TypeSpec) (string, error) {
+	typ := prop.Type
+	if typ != "" && typ != "object" && typ != "array" {
```

Comment:
Need to check for "ref" here?

Reply:
I don't think so, since `Type` shouldn't be set in that case.  I added a check for `&& prop.Ref == ""` now though.